hadoop-plugin/build.gradle

```diff
@@ -13,20 +13,38 @@ repositories {
 // The group will be overriden in LinkedIn builds
 group=rootProject.group
 
+configurations {
+  provided
+}
+
+sourceSets {
+  main {
+    compileClasspath+=configurations.provided
+  }
+  test {
+    compileClasspath+=configurations.provided
+    runtimeClasspath+=configurations.provided
+  }
+}
+
 dependencies {
   compile gradleApi()
   compile localGroovy()
   compile project(":oozie-jaxb")
 
   compile 'javax.xml.bind:jaxb-api:2.2.7'
-  compile('org.apache.hadoop:hadoop-auth:2.3.0')   { transitive = false }
-  compile('org.apache.hadoop:hadoop-common:2.3.0') { transitive = false }
-  compile('org.apache.hadoop:hadoop-hdfs:2.3.0')   { transitive = false }
+  compile('org.apache.hadoop:hadoop-auth:2.3.0')
```

Comment:
We need to keep transitive=false for the Hadoop dependencies. Otherwise people trying to use the Hadoop Plugin will download hundreds of MB's in Hadoop jars the first time they try to do something. If you are missing a jar that you need at runtime, please add it directly as a runtime dependency